bingo.js

```diff
@@ -240,38 +255,50 @@ function loadSettings()
 {
 	getSettingsFromURL();
 	getSettingsFromLocalStorage();
+	checkRecentCurrentSheet();
+	loadProgressFromURL();
 }
 
-function getSettingsFromURL()
+function getSettingsFromURLSplitted(url)
 {
 	/**
 	 * URL Format: ?s=[difficulty]-[hideTable]_[seed]
 	 *
 	 * The seed should always be last, so in order to be able to add more settings,
 	 * settings and the seed are separated from eachother.
 	 */
-	var split = gup("s").split("_");
-	if (split.length == 2)
-	{
-		var settings = split[0].split("-");
-		SEED = split[1];
+	 var seed, difficulty, version, hidden, streamerMode;
 
-		DIFFICULTY = parseInt(settings[0]);
-		HIDDEN = settings[1] == "1";
-		STREAMER_MODE = settings[2] == "1";
-		var selectedVersion = settings[3];
-	}
+	 if (url === undefined) url = gup("s");
+	 var split = url.split("_");
+	 if (split.length == 2)
+	 {
+		 var settings = split[0].split("-");
+		 seed = split[1];
 
-	// Set default values
-	if (isNaN(DIFFICULTY) || DIFFICULTY < 1 || DIFFICULTY > 5)
-	{
-		DIFFICULTY = 3;
-	}
+		 difficulty = parseInt(settings[0]);
+		 hidden = settings[1] == "1";
+		 streamerMode = settings[2] == "1";
+		 var selectedVersion = settings[3];
+	 }
```

Comment:
Code style nitpick: this whole function is indented with an extra unnecessary space.